tests/unit/fancyrag/qa/test_evaluator.py

```
@@ -1135,4 +1011,585 @@ def _chunk_record():
+    """Test _estimate_tokens rounds up character count / 4."""
+    # 4 chars = 1 token
+    assert IngestionQaEvaluator._estimate_tokens("abcd") == 1
+    # 5 chars = 2 tokens (rounded up)
+    assert IngestionQaEvaluator._estimate_tokens("abcde") == 2
+    # 8 chars = 2 tokens
+    assert IngestionQaEvaluator._estimate_tokens("a" * 8) == 2
+    # 9 chars = 3 tokens (rounded up)
+    assert IngestionQaEvaluator._estimate_tokens("a" * 9) == 3
+
+
+def test_bucket_label_formatting():
```

Ruff (0.13.3)

Redefinition of unused `test_bucket_label_formatting` from line 649

(F811)